src/haddock/modules/scoring/sasascore/sasascore.py

```diff
@@ -0,0 +1,165 @@
+"""Accessibility scoring calculations."""
+from haddock import log
+from pathlib import Path
+import pandas as pd
+from haddock.core.typing import FilePath
+from haddock.clis.restraints.calc_accessibility import (
+    apply_cutoff,
+    get_accessibility,
+    )
+from typing import Optional
+
+
+def prettify_df(output_fname: str, columns: list[str], sortby: Optional[str] = None) -> None:
+    """Prettify the output dataframe.
+
+    Parameters
+    ----------
+    output_fname : str
+        The name of the output file.
+    
+    columns : list
+        The columns of the dataframe.
+    
+    sortby : str
+        The column to sort by.
+    """
+    # dataframe conversion and sorting
+    df = pd.read_csv(output_fname, sep="\t", header=None)
+    df.columns = columns
+    if sortby:
+        df = df.sort_values(by=sortby, ascending=True)
+    df.to_csv(output_fname, sep="\t", index=False, na_rep="None", float_format="%.0f")
+    log.info(f"{output_fname} created.")
+
+
+def calc_acc_score(result_dict, buried_resdic, acc_resdic):
+    """Calculate the accessibility score.
+    
+    The accessibility score is calculated as the sum of the number of
+    residues that are not in the correct category (buried or accessible).
+    
+    Parameters
+    ----------
+    result_dict : dict
+        A dictionary with the results from the accessibility calculation.
+        
+    buried_resdic : dict
+        A dictionary with the buried residues.
+    
+    acc_resdic : dict
+        A dictionary with the accessible residues.
+    
+    Returns
+    -------
+    acc_score : int
+        The accessibility score.
+    """
+    # filter the residues
+    acc_score: int = 0
+    b_viols: dict[str, set] = {}
+    a_viols: dict[str, set] = {}
+    for ch in result_dict:
+        if ch in buried_resdic:
+            # for every supposedly buried residue that is not buried,
+            # the score should increase by one
+            b_viols[ch] = set(buried_resdic[ch]).difference(result_dict[ch])
+            buried_viol_sc = len(b_viols[ch])
+            acc_score += buried_viol_sc
+        if ch in acc_resdic:
+            # now the opposite logic with the accessible amino acids.
+            a_viols[ch] = set(acc_resdic[ch]).difference(result_dict[ch])
+            acc_viol_sc = len(a_viols[ch])
+            acc_score += acc_viol_sc
+    return acc_score, b_viols, a_viols
+
+
+class AccScoreJob:
+    """A Job dedicated to the parallel running of Accscore jobs."""
+
+    def __init__(
+            self,
+            accscore_obj):
+        """Initialise AccScoreJob."""
+        self.accscore_obj = accscore_obj
+        self.output = accscore_obj.output_name
+
+    def run(self) -> None:
+        """Run this AccScoreJob."""
+        log.info(f"core {self.accscore_obj.core}, running AccScore...")
+        self.accscore_obj.run()
+        self.accscore_obj.output()
+        return
```

Comment:
Why this?

Reply:
?

Reply:
Why do you have two classes with two run methods; `AccScoreJob` and `AccScore`?